hail/src/main/scala/is/hail/expr/ir/analyses/SemanticHash.scala

```diff
@@ -91,8 +91,18 @@ case object SemanticHash extends Logging {
       case a: AggGroupBy =>
         buffer += a.isScan.toByte
 
-      case a: AggLet =>
-        buffer += a.isScan.toByte
+      case Block(bindings, _) =>
+        // encode scopes
+        for {
+          group <- bindings.map(_.scope).grouped(4)
+        } {
+          var byte: Int = 0
+          for (i <- group) {
+            byte <<= 2
+            byte |= i
+          }
```

Comment:
Why? Why not just leave it as 4 ints?
```
buffer ++= bindings.map(_.scope).grouped(4).map(_.toByte)
```

Reply:
Sure, I could do that until we have evidence this is a hot spot. I'm just anticipating blocks will get quite large.